### ISSUE TITLE
Two foreign keys between two tables generate duplicate join methods

### ISSUE THREAD

**Issue by MeanSquaredError** [stephenafamo/bob#16]
I have the following two PostgreSQL tables:
```
CREATE TABLE tg_message (
	channel_id int8 NOT NULL REFERENCES tg_channel(channel_id),
	message_id int NOT NULL,
	message_ts int NOT NULL,
	body text NOT NULL,

	PRIMARY KEY (channel_id, message_id)
);
CREATE INDEX tg_message_channel_id ON tg_message(channel_id);

CREATE TABLE tg_gap (
	gap_id SERIAL PRIMARY KEY,
	channel_id int8 NOT NULL REFERENCES tg_channel(channel_id),
	msg_id_min int,
	msg_id_max int,
	last_attempt_ts int8,

	FOREIGN KEY (channel_id, msg_id_min) REFERENCES tg_message (channel_id, message_id),
	FOREIGN KEY (channel_id, msg_id_max) REFERENCES tg_message (channel_id, message_id)
);
```

bobgen-psql generates the following duplicate join methods:
```
func tgGapsJoinTGMessages[Q dialect.Joinable](ctx context.Context, typ string) bob.Mod[Q] {
	return mods.QueryMods[Q]{
		dialect.Join[Q](typ, TGGapsTable.Name(ctx)).On(
			TGMessageColumns.MessageID.EQ(TGGapColumns.MSGIDMax), TGMessageColumns.ChannelID.EQ(TGGapColumns.ChannelID), TGMessageColumns.ChannelID.EQ(TGGapColumns.MSGIDMax), TGMessageColumns.MessageID.EQ(TGGapColumns.ChannelID),
		),
	}
}
func tgGapsJoinTGMessages[Q dialect.Joinable](ctx context.Context, typ string) bob.Mod[Q] {
	return mods.QueryMods[Q]{
		dialect.Join[Q](typ, TGGapsTable.Name(ctx)).On(
			TGMessageColumns.MessageID.EQ(TGGapColumns.MSGIDMin), TGMessageColumns.ChannelID.EQ(TGGapColumns.ChannelID), TGMessageColumns.ChannelID.EQ(TGGapColumns.MSGIDMin), TGMessageColumns.MessageID.EQ(TGGapColumns.ChannelID),
		),
	}
}
```

I can see two problems here:

1. The join condition seems broken, more specifically these two join conditions look incorrect to me:
```
TGMessageColumns.ChannelID.EQ(TGGapColumns.MSGIDMin), TGMessageColumns.MessageID.EQ(TGGapColumns.ChannelID)
```

2. There are two identical join methods `tgGapsJoinTGMessages`, which is a problem.

**Comment by stephenafamo** [comment]
Looking into this

**Comment by stephenafamo** [comment]
### The join condition seems broken, more specifically these two join conditions look incorrect to me:

There was a bug when joining tables to retrieve foreign keys. This is fixed now on the `main` branch. I'll tag a new release soon.

### There are two identical join methods

This is because Bob generates identical names for both relationships. In this case you should define relationship aliases for them explicitly. [See Documentation](https://bob.stephenafamo.com/docs/code-generation/configuration#aliases)

**Comment by MeanSquaredError** [comment]
I tested bobgen-psql from the main branch and the first bug (incorrect join conditions) is solved. Thank very much you for the quick bugfix!

The second issue was solved by using a foreign key alias. Thank you for providing a link to the documentation.

It probably makes sense to expand a bit the documentation on foreign key aliases, explaining that the foreign key alias changes the name of the foreign table only, not the whole name of the foreign key.

**Comment by stephenafamo** [comment]
Released the fix in `v0.17.2`

I'll work on improving the documentation.

**Comment by MeanSquaredError** [comment]
By the way, it is probably worth mentioning in the documentation that having two foreign keys between two tables means that there will be duplicate structure elements and/or methods on both sides on the foreign key. In my case I had duplicates both for the tg_gap table and for the tg_message table. So in this case I had to add two aliases in the configuration yaml file like this:

```
aliases:
  tables:
    tg_gap:
      relationships:
        tg_gap_channel_id_msg_id_min_fkey: TGMessage2
    tg_message:
      relationships:
        tg_gap_channel_id_msg_id_min_fkey: TGGaps2
```
